vllm/model_executor/models/phi3v.py

```diff
@@ -402,55 +403,84 @@ def input_processor_for_phi3v(ctx: InputContext, llm_inputs: LLMInputs):
         w, h = image_data.size
         w, h = _calc_hd_transform_size(width=w, height=h)
 
-        image_feature_size = get_phi3v_image_feature_size(hf_config,
-                                                          input_width=w,
-                                                          input_height=h)
+        image_feature_size = [
+            get_phi3v_image_feature_size(hf_config,
+                                         input_width=w,
+                                         input_height=h)
+        ]
+        image_data = [image_data]
+    elif is_list_of(image_data, Image.Image):
+        image_feature_size = []
+        for image in image_data:
+            w, h = image.size
+            w, h = _calc_hd_transform_size(width=w, height=h)
+            image_feature_size.append(
+                get_phi3v_image_feature_size(hf_config,
+                                             input_width=w,
+                                             input_height=h))
     elif isinstance(image_data, torch.Tensor):
         image_feature_size = image_data.shape[0]
     else:
         raise TypeError(f"Invalid image type: {type(image_data)}")
 
     prompt = llm_inputs.get("prompt")
     if prompt is None:
+        image_idx = []
         new_prompt = None
     else:
+        image_idx = sorted(
+            list(map(int, re.findall(r"<\|image_(\d+)\|>+", prompt))))
```

Comment:
```suggestion
        image_idx = sorted(
            map(int, re.findall(r"<\|image_(\d+)\|>+", prompt)))
```

`sorted` accepts an iterable so we don't need to convert it into a list beforehand.